legend-engine-ide-lsp-server/src/main/java/org/finos/legend/engine/ide/lsp/server/LegendTextDocumentService.java

```diff
@@ -157,42 +167,41 @@ public void didChange(DidChangeTextDocumentParams params)
     @Override
     public void didClose(DidCloseTextDocumentParams params)
     {
-        this.server.runPossiblyAsync(() ->
+        LegendServerGlobalState globalState = this.server.getGlobalState();
+        this.server.checkReady();
```

Comment:
I don't think it's necessary to call checkReady here, as that's done in getGlobalState.